indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -444,8 +445,7 @@ public TaskStatus run(final TaskToolbox toolbox)
             toolbox.getTaskActionClient(),
             intervals
         );
-        versions = locks.entrySet().stream()
-                        .collect(Collectors.toMap(Entry::getKey, entry -> entry.getValue().getVersion()));
+        versions = CollectionUtils.mapValues(locks, v -> v.getVersion());
```

Comment:
It's more readable to use method references in such cases because it makes evident the type of the value: `versions = CollectionUtils.mapValues(locks, TaskLock::getVersion);`. Or, it's better to choose more descriptive lambda param names (e. g. `taskLock`) than just `v`.

Same applies to many other places in this PR.